packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -829,8 +851,17 @@ function buildTree(
   const stackOfChildren = [];
   for (let i = 0; i < readHookLog.length; i++) {
     const hook = readHookLog[i];
-    const stack = parseTrimmedStack(rootStack, hook);
-    if (stack !== null) {
+    const parseResult = parseTrimmedStack(rootStack, hook);
+    let displayName = hook.displayName;
+    if (parseResult !== null) {
+      const [primitiveFrame, stack] = parseResult;
+      if (hook.displayName === null) {
+        displayName =
+          parseHookName(primitiveFrame.functionName) ||
+          // Older versions of React do not have sourcemaps.
+          // In those versions there was always a 1:1 mapping between wrapper and dispatcher method.
+          parseHookName(hook.dispatcherMethodName);
+      }
```

Comment:
```suggestion
        displayName =
          parseHookName(primitiveFrame.functionName) ||
          // Older versions of React do not have sourcemaps.
          // In those versions there was always a 1:1 mapping between wrapper and dispatcher method.
          parseHookName(hook.dispatcherMethodName) ||
          hook.primitive;
      }
```

Reply:
At least we will be showing `primitive` name instead of `Unknown`, like it was before.

I've observed this with `CustomHooks` element in `react-devtools-shell`

Reply:
Good catch. This should be fixed by though: [`3deebb2` (#28593)](https://github.com/facebook/react/pull/28593/commits/3deebb2b3fb66951bf9d479903d5cc6b3778a3ac). I'll get your suggestion in is as well since it's a nicer fallback.